browsers/duckduckgo.safariextension/js/background.js

```diff
@@ -106,4 +107,59 @@ var onBeforeRequest = function (requestData) {
     }   
 }
 
+
+/**
+ * Before navigating to a new page,
+ * check whether we should upgrade to https
+ */
+var onBeforeNavigation = function (e) {
+    //console.log("beforeNavigation", e.url, e.target.url, e.eventPhase, e)
+
+    if (!e.url) return
+
+    const tabId = tabManager.getTabId(e)
+    let thisTab = tabManager.get({tabId: tabId})
+
+    if (!thisTab) {
```

Comment:
What case does this happen?

Reply:
Most of them I think, `beforeNavigation` fires before the other `canLoad` event handler above where the tab was previously being created.